AwesomeProject/App.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { Component } from 'react';
+import { Text, View, TextInput, Button } from 'react-native';
+
+interface IState {
+  email?: string;
+  password?: string;
+  errorMessage?: string;
+}
+
+interface IProps {}
+
+export default class HelloWorldApp extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props);
+    this.state = {email:"", password:"", errorMessage:""};
+  }
+
+  validateEmailAndPassword(): void {
+    const regexEmail:RegExp = /([A-Za-z])+@([A-Za-z])+.com/gm;
+    const regexOneCharAndOneDigit:RegExp = /(?=.*?[0-9])(?=.*?[A-Za-z]).+/gm;
+    if (this.state.email.length == 0) {
```

Comment:
Se você usar só `if (this.state.email) {`, ele já faz algumas checagens até mais abrangentes do que a sua. Essa 
expressão avalia para false caso `this.state.email` seja `undefined`, `null` ou uma string vazia mesmo. Mais infos [aqui](https://javascript.info/ifelse)

Reply:
O mesmo vale para o método que avalia a senha.

Reply:
Muito legal essa propriedade! ;) Corrigido